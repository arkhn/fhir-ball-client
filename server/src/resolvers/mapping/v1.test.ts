import { Photon } from '@prisma/photon'
import set from 'lodash.set'

import importMappingV1 from './v1'
import * as mappingV1 from '../../../test/fixtures/chimio-mapping-v1.json'

const mockCreateResource = jest.fn()
jest.mock('@prisma/photon', () => ({
  Photon: jest.fn().mockImplementation(() => ({
    resources: {
      create: (data: any) => mockCreateResource(data),
    },
  })),
}))

describe('import mapping V1', () => {
  const sourceId = '01234567'
  const resourceCount = 2
  const { resources } = mappingV1 as any

  beforeEach(() => {
    mockCreateResource.mockClear()
  })

  it('should flatten the mapping and send a query per resource', async () => {
    await importMappingV1(new Photon(), sourceId, resources)
    expect(mockCreateResource).toHaveBeenCalledTimes(resourceCount)
    expect(mockCreateResource.mock.calls[0]).toMatchSnapshot() // EpisodeOfCare - HopitalStay
    expect(mockCreateResource.mock.calls[1]).toMatchSnapshot() // Patient
  })

  it('should have cleaned the resource and attributes', async () => {
    await importMappingV1(new Photon(), sourceId, resources)
    expect(mockCreateResource.mock.calls[0][0]).toEqual({
      data: {
        label: resources[0].label,
        primaryKeyOwner: resources[0].primaryKeyOwner,
        primaryKeyTable: resources[0].primaryKeyTable,
        primaryKeyColumn: resources[0].primaryKeyColumn,
        source: {
          connect: {
            id: sourceId,
          },
        },
        definitionId: 'EpisodeOfCare',
        attributes: {
          create: expect.arrayContaining([
            {
              path: 'period.start',
              comments: null,
              definitionId: '',
              mergingScript: 'merge_concat',
              inputs: {
                create: expect.any(Array),
              },
            },
          ]),
        },
      },
    })
  })

  it('should have computed the path correctly', async () => {
    await importMappingV1(new Photon(), sourceId, resources)
    const {
      attributes: { create: attributes },
    } = mockCreateResource.mock.calls[1][0].data // Patient attributes
    let structure = {}
    for (const a of attributes) {
      set(structure, a.path, a.inputs!.create) // use lodash to fill a structured object
    }
    expect(structure).toMatchSnapshot()
  })
})
