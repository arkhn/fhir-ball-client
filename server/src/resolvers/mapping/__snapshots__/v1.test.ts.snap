// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping V1 should flatten the mapping and send a query per resource 1`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.reference",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SERVICE",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ETATSEJ",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "status",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SECLEUNIK",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION2",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "select_first_not_empty",
            "path": "identifier[1].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TWCLEUNIK",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].code",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "LIBELTW",
                      "joins": Object {
                        "create": Array [
                          Object {
                            "tables": Object {
                              "create": Array [
                                Object {
                                  "column": "TWCLEUNIK",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "SEJOUR",
                                },
                                Object {
                                  "column": "TWCLEUNIK",
                                  "owner": "OPS$ACHQ1ABC",
                                  "table": "TYPESEJ",
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "owner": "OPS$ACHQ1ABC",
                      "table": "TYPESEJ",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].display",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREOUT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.end",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEIN",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREIN",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.start",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.reference",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "/Patient/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.system",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.value",
          },
        ],
      },
      "definitionId": "EpisodeOfCare",
      "label": "Hospital Stay",
      "primaryKeyColumn": "SECLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "SEJOUR",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;

exports[`import mapping V1 should flatten the mapping and send a query per resource 2`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "active",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR1PAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR2PAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR3PAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "address[0].line[0]",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "VILLEPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].city",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CPOSTPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].postalCode",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "IDMOTIFINAC",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedBoolean",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINACPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedDateTime",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TELPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "telecom[0].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "official",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].use",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].family",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PRENOMPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].given[0]",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "maiden",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].use",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMJFPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].family",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATENAISPA",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "birthDate",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SEXEPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "gender",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "http://terminology.interopsante.org/CodeSystem/v2-0203",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].system",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "PI",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].code",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "Patient internal identifier",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].display",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "false",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].userSelected",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOOBSPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[1].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DOSSIERPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].value",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.start",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUTPAT",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.end",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "/Organization",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.type",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "comments": null,
            "definitionId": "",
            "inputs": Object {
              "create": Array [
                Object {
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CODESERV",
                      "owner": "OPS$ACHQ1ABC",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
        ],
      },
      "definitionId": "Patient",
      "label": "",
      "primaryKeyColumn": "PACLEUNIK",
      "primaryKeyOwner": "OPS$ACHQ1ABC",
      "primaryKeyTable": "PATIENT",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;

exports[`import mapping V1 should have computed the path correctly 1`] = `
Object {
  "active": Array [
    Object {
      "script": null,
      "sqlValue": Object {
        "create": Object {
          "column": "INACTIVPAT",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
  ],
  "address": Array [
    Object {
      "city": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "VILLEPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
      "line": Array [
        Array [
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "ADR1PAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "ADR2PAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "ADR3PAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
        ],
      ],
      "postalCode": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "CPOSTPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
  ],
  "birthDate": Array [
    Object {
      "script": "clean_date",
      "sqlValue": Object {
        "create": Object {
          "column": "DATENAISPA",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
  ],
  "deceasedBoolean": Array [
    Object {
      "script": null,
      "sqlValue": Object {
        "create": Object {
          "column": "INACTIVPAT",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
    Object {
      "script": null,
      "sqlValue": Object {
        "create": Object {
          "column": "IDMOTIFINAC",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
  ],
  "deceasedDateTime": Array [
    Object {
      "script": null,
      "sqlValue": Object {
        "create": Object {
          "column": "DATEINACPAT",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
  ],
  "gender": Array [
    Object {
      "script": null,
      "sqlValue": Object {
        "create": Object {
          "column": "SEXEPAT",
          "owner": "OPS$ACHQ1ABC",
          "table": "PATIENT",
        },
      },
      "staticValue": null,
    },
  ],
  "identifier": Array [
    Object {
      "type": Object {
        "coding": Array [
          Object {
            "code": Array [
              Object {
                "script": null,
                "staticValue": "PI",
              },
            ],
            "display": Array [
              Object {
                "script": null,
                "staticValue": "Patient internal identifier",
              },
            ],
            "system": Array [
              Object {
                "script": null,
                "staticValue": "http://terminology.interopsante.org/CodeSystem/v2-0203",
              },
            ],
            "userSelected": Array [
              Object {
                "script": null,
                "staticValue": "false",
              },
            ],
          },
        ],
      },
      "value": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "NOOBSPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
    Object {
      "value": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "PACLEUNIK",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
    Object {
      "period": Object {
        "end": Array [
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "DATEOUTPAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
        ],
        "start": Array [
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "DATEINPAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
        ],
      },
      "value": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "DOSSIERPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
  ],
  "managingOrganization": Object {
    "identifier": Object {
      "system": Array [
        Object {
          "script": null,
          "staticValue": "/Organization/{id}",
        },
      ],
      "value": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "CODESERV",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
    "type": Array [
      Object {
        "script": null,
        "staticValue": "/Organization",
      },
    ],
  },
  "name": Array [
    Object {
      "family": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "NOMPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
      "given": Array [
        Array [
          Object {
            "script": null,
            "sqlValue": Object {
              "create": Object {
                "column": "PRENOMPAT",
                "owner": "OPS$ACHQ1ABC",
                "table": "PATIENT",
              },
            },
            "staticValue": null,
          },
        ],
      ],
      "use": Array [
        Object {
          "script": null,
          "staticValue": "official",
        },
      ],
    },
    Object {
      "family": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "NOMJFPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
      "use": Array [
        Object {
          "script": null,
          "staticValue": "maiden",
        },
      ],
    },
  ],
  "telecom": Array [
    Object {
      "value": Array [
        Object {
          "script": null,
          "sqlValue": Object {
            "create": Object {
              "column": "TELPAT",
              "owner": "OPS$ACHQ1ABC",
              "table": "PATIENT",
            },
          },
          "staticValue": null,
        },
      ],
    },
  ],
}
`;
